pyupgrade/_plugins/open_mode.py

```diff
@@ -48,11 +49,21 @@ def visit_Call(
             state.settings.min_version >= (3,) and
             isinstance(node.func, ast.Name) and
             node.func.id == 'open' and
-            not has_starargs(node) and
-            len(node.args) >= 2 and
-            isinstance(node.args[1], ast.Str) and (
+            not has_starargs(node)
+    ):
+        if len(node.args) >= 2 and isinstance(node.args[1], ast.Str):
+            if (
                 node.args[1].s in U_MODE_REPLACE or
                 (len(node.args) == 2 and node.args[1].s in U_MODE_REMOVE)
-            )
-    ):
-        yield ast_to_offset(node), _fix_open_mode
+            ):
+                yield ast_to_offset(node), _fix_open_mode
+        elif node.keywords:
+            mode = [i.value for i in node.keywords if i.arg == 'mode']
+            if (
+                mode and
+                isinstance(mode[0], ast.Str) and
+                mode[0].s in U_MODE_REMOVE.union(U_MODE_REPLACE)
```

Comment:
this builds a set just to check containment (turns 2 O(1) operations into a slow O(N) operation)